api/service/prometheus/service.go

```diff
@@ -0,0 +1,122 @@
+// Package prometheus defines a service which is used for metrics collection
+// and health of a node in Harmony.
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"runtime/debug"
+	"runtime/pprof"
+	"time"
+
+	nodeconfig "github.com/harmony-one/harmony/internal/configs/node"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog"
+)
+
+// Service provides Prometheus metrics via the /metrics route. This route will
+// show all the metrics registered with the Prometheus DefaultRegisterer.
+type Service struct {
+	server     *http.Server
+	failStatus error
+}
+
+// Handler represents a path and handler func to serve on the same port as /metrics, /healthz, /goroutinez, etc.
+type Handler struct {
+	Path    string
+	Handler func(http.ResponseWriter, *http.Request)
+}
+
+var (
+	svc = &Service{}
+)
+
+// NewService sets up a new instance for a given address host:port.
+// An empty host will match with any IP so an address like ":19000" is perfectly acceptable.
+func NewService(config nodeconfig.PrometheusServerConfig, additionalHandlers ...Handler) {
```

Comment:
I would suggest to define PrometheusServerConfig in this module instead of in nodeconfig.

Reply:
we use node config for all the configuration, I think that's easier to see and change.